src/flask/helpers.py

```diff
@@ -340,10 +340,14 @@ def external_url_handler(error, endpoint, values):
     except BuildError as error:
         # We need to inject the values again so that the app callback can
         # deal with that sort of stuff.
-        values["_external"] = external
-        values["_anchor"] = anchor
-        values["_method"] = method
-        values["_scheme"] = scheme
+        values.update(
+            {
+                "_external": external,
+                "_anchor": anchor,
+                "_method": method,
+                "_scheme": scheme,
+            }
+        )
```

Comment:
This is longer and doesn't look better. The usecase for `dict.update()` is when you already have a sequence...